dev/tools/gen_keycodes/README.md

```diff
@@ -58,109 +58,62 @@ the API.
 
-The logical key code is a 37-bit integer in a namespace that we control and
-define. It has values in the following ranges.
+The logical key code is a 52-bit integer (due to the limitation of JavaScript).
+The entire namespace is divided into 32-bit *planes*. The upper 20 bits of the
+ID represent the plane ID, while the lower 32 bits represent values in the
+plane.  For example, plane 0x1 refers to the range 0x1 0000 0000 -
+0x1 FFFF FFFF.  Each plane manages how the values within the range are assigned.
 
-- **0x00 0000 0000 - 0x0 0010 FFFF**: For keys that generate Unicode
+The planes are planned as follows:
+
+- **Plane 0x00**: The Unicode plane. This plane contains keys that generate Unicode
   characters when pressed (this includes dead keys, but not e.g. function keys
-  or shift keys), the logical key code is the Unicode code point corresponding
-  to the representation of the key in the current keyboard mapping. The
-  Unicode code point might not match the string that is generated for
-  an unshifted keypress of that key, for example, we would use U+0034 for the
-  “4 \$” key in the US layout, and also the “4 ;” key in the Russian layout,
-  and also, maybe less intuitively, for the “' 4 {“ in French layout (wherein
-  the latter case, an unshifted press gets you a ', not a 4). Similarly, the Q
-  key in the US layout outputs a q in normal usage, but its code would be 0x0
-  0000 0051 (U+00051 being the code for the uppercase Q).
-
-- **0x01 0000 0000 - 0x01 FFFF FFFF**: For keys that are defined by the [USB HID
-  standard](https://www.usb.org/sites/default/files/documents/hut1_12v2.pdf),
-  the key code consists of the 32 bit USB extended usage code. For
-  example, the Enter key would have code 0x01 0007 0028. Only keys that fall
-  into collections "Keyboard", "Keypad", and "Tablet PC System Controls" are
-  considered for this API; for example, a mixing desk with multiple
-  collections of volume controls would not be exposed via DOWN and UP events,
-  nor would a mouse, joystick, or golf simulator control.
-
-- **0x02 0000 0000 - 0xFF FFFF FFFF**: For keys that aren't defined in USB at the
-  time of implementation, but that we need to support. For example, if Flutter
-  were ever ported to the Symbolics LM-2, the "thumb up" key might be given
-  the code 0x14 0000 0001, where 0x14 is defined as the “Symbolics” platform
-  range. Where possible, we will use specific subranges of this space to reuse
-  keys from other platforms. When this is not possible, the prefix 0xFF is
-  reserved for “Custom” codes. Each platform from which we take codes will get
-  a unique prefix in the range 0x2-0xFE. If multiple systems define keys with
-  the same usage (not the same number), then the value with the lowest prefix
-  is used as the defining code.
-
-  Prefixes will be:
+  or shift keys). The value is defined as the Unicode code point corresponding
+  to the character, lower case and without modifier keys if possible.
```

Comment:
First, this means the exact opposite: all special characters have their own key. The "+", the ".", the "@". Also, EASCII letters have their own key as lower case too: the "ü" key. 

The problems is how we list these keys. 

First, symbol keys have to be listed separated from their non-shift keys. On Web Shift-2 on US keyboard will only provide `key: "@"` and you'll never know it comes from a "2" key unless from `code` (physical). This apparently applies to macOS too, since they have `characters` and `charactersWithoutModifiers`; I think it applies to GTK too, since they explicitly defined letter keys for both cases, as well as all symbols.

(However, this is not universal either. Win32 gives you `key: 0x32` (the 2 key) in both "2" and "Shift-2" on a US keyboard. It also gives the same `key` in both cases on a French keyboard. Although, they can be told apart by the character, so we probably should derive the logical key using the character field.)

So the problem becomes: should we combine upper letters with lower letters for logical keys? I agree that combining them makes the logical key definition more confusing, since they're basically the only keys that "hide Shift". However, separating them also makes shortcuts confusing and harder: if you want to change the code that defines "Ctrl-A" to "Ctrl-Shift-A", you'll have to remember changing the `loweA` to `upperA` or it will never trigger.

One more disadvantage for separating the letter keys is a slight breakage: currently we only have `keyA`. If we separate letter keys, they'll probably have to be renamed to `lowerA` and `upperA` (although we can keep both).

Reply:
Through local discussion we've decided to move forward with this change. It'll be implemented in a separate PR.